webpack/components/AnsibleRoles/AnsibleRolesPage.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import TableIndexPage from 'foremanReact/components/PF4/TableIndexPage/TableIndexPage';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { getDocsURL } from 'foremanReact/common/helpers';
+import { AnsibleRolesImportButton } from './components/AnsibleRolesImportButton';
+import { getActions } from '../AnsibleRoles/AnsibleRolesPageHelpers';
+
+export const AnsibleRolesPage = () => (
+  <TableIndexPage
+    header={__('Ansible Roles')}
+    controller="ansible_roles"
+    apiUrl="/ansible/ui_ansible_roles"
+    apiOptions={{ key: 'WEBHOOKS_API_REQUEST_KEY' }}
+    hasHelpPage
+    creatable={false}
+    isDeleteable
+    searchable
+    customToolbarItems={
+      <AnsibleRolesImportButton>Primary</AnsibleRolesImportButton>
+    }
+    customHelpURL={getDocsURL(
+      'Managing_Configurations_Ansible',
+      'Importing_Ansible_Roles_and_Variables_ansible'
+    )}
+    rowKebabItems={getActions}
+    columns={{
+      import_time: { title: __('Imported at'), isSorted: true },
+      variables_count: { title: __('Variables') },
+      hosts_count: { title: __('Hosts') },
+      hostgroups_count: { title: __('Hostgroups') },
+      name: { title: __('Name'), isSorted: true },
```

Comment:
Do we want to use a new order, or should this be re-ordered to fit the old page?

Reply:
For me, this arrangement orders the columns the same way they have been on the old version.

Reply:
the old version is arranged opposite way, with name being the first/most left, or am I missing something?
 ```
 <th class="col-md-6"><%= sort :name, :as => s_("Role|Name") %></th>
      <th class="col-md-2"><%= _("Hostgroups") %></th>
      <th class="col-md-2"><%= _("Hosts") %></th>
      <th class="col-md-2"><%= _("Variables") %></th>
      <th class="col-md-2"><%= sort :updated_at, :as => _("Imported at") %></th>
      <th class="col-md-2"><%= _("Actions") %></th>
```
      
![image](https://github.com/theforeman/foreman_ansible/assets/30431079/854e6f62-a22d-48a1-806b-61aa1a2e570b)
![image](https://github.com/theforeman/foreman_ansible/assets/30431079/894f50e0-85f5-4c67-b610-dc920b8ec265)

Reply:
Now this I don't understand...
![image](https://github.com/theforeman/foreman_ansible/assets/99347625/24f6a45d-e5e4-44fe-8fbd-965042ae9425)

Reply:
This is fun, I'll try to see why it happens next week

Reply:
Looks like firefox and chrome implement `sort` differently, I'll find a fix in foreman core. so this is fine

Reply:
Can you add weight key to each title?  just to ease the foreman core fix. Lower weights are displayed first.

Reply:
Ah yes, I looked into this a bit as well.
Apparently, ES2019 just mandates that the sorting algorithm must be stable, but the actual implementation of `sort` is up to the developers. 

When looking at the sorting function in [getColumnHelpers of helpers.js](https://github.com/theforeman/foreman/blob/8283db898b7e145e59dd376ac0b4cb2d30f7b436/webpack/assets/javascripts/react_app/components/PF4/TableIndexPage/Table/helpers.js#L26C1-L44C3) with a debugger, one can see that the order in which the items in the array are consumed differs between FireFox and Chromium based browsers:

FireFox:

1.  a: "import_time"
b: "variables_count"
2. a: "variables_count"
b: "hosts_count"
3. a: "hosts_count"
b: "hostgroups_count"
4. a: "hostgroups_count"
b: "name"

Chromium:

1.  a: "variables_count"
b: "import_time"
2. a: "hosts_count"
b: "variables_count"
3. a: "hostgroups_count"
b: "hosts_count"
4. a: "name"
b: "hostgroups_count"

This can be reproduced by this simple example:
```js
const test_array = ["a", "b", "c", "d", "e"];
console.log(test_array.sort((a, b) => 1));
```

A fix could be to have the sorting function return `0` if no weights are applied, as that produces the same order as the unsorted array.
Either way, adding weights to the columns solved this on both platforms.